rust/libceed/src/operator.rs

```diff
@@ -20,18 +20,292 @@
 
 use crate::prelude::*;
 
+// -----------------------------------------------------------------------------
+// CeedOperator Field context wrapper
+// -----------------------------------------------------------------------------
+#[derive(Debug)]
+pub struct OperatorField<'a> {
+    ceed: &'a crate::Ceed,
+    pub(crate) ptr: bind_ceed::CeedOperatorField,
+}
+
+// -----------------------------------------------------------------------------
+// Implementations
+// -----------------------------------------------------------------------------
+impl<'a> OperatorField<'a> {
+    /// Get the name of an OperatorField
+    ///
+    /// ```
+    /// # use libceed::prelude::*;
+    /// # let ceed = libceed::Ceed::default_init();
+    /// let qf = ceed.q_function_interior_by_name("Mass1DBuild").unwrap();
+    ///
+    /// // Operator field arguments
+    /// let ne = 3;
+    /// let q = 4 as usize;
+    /// let mut ind: Vec<i32> = vec![0; 2 * ne];
+    /// for i in 0..ne {
+    ///     ind[2 * i + 0] = i as i32;
+    ///     ind[2 * i + 1] = (i + 1) as i32;
+    /// }
+    /// let r = ceed
+    ///     .elem_restriction(ne, 2, 1, 1, ne + 1, MemType::Host, &ind)
+    ///     .unwrap();
+    /// let strides: [i32; 3] = [1, q as i32, q as i32];
+    /// let rq = ceed
+    ///     .strided_elem_restriction(ne, 2, 1, q * ne, strides)
+    ///     .unwrap();
+    ///
+    /// let b = ceed
+    ///     .basis_tensor_H1_Lagrange(1, 1, 2, q, QuadMode::Gauss)
+    ///     .unwrap();
+    ///
+    /// // Operator fields
+    /// let op = ceed
+    ///     .operator(&qf, QFunctionOpt::None, QFunctionOpt::None)
+    ///     .unwrap()
+    ///     .field("dx", &r, &b, VectorOpt::Active)
+    ///     .unwrap()
+    ///     .field("weights", ElemRestrictionOpt::None, &b, VectorOpt::None)
+    ///     .unwrap()
+    ///     .field("qdata", &rq, BasisOpt::Collocated, VectorOpt::Active)
+    ///     .unwrap();
+    ///
+    /// let inputs = op.inputs().unwrap();
+    ///
+    /// assert_eq!(inputs[0].name(), "dx", "Incorrect input name");
+    /// assert_eq!(inputs[1].name(), "weights", "Incorrect input name");
+    /// ```
+    pub fn name(&self) -> String {
+        let mut name_ptr: *mut std::os::raw::c_char = std::ptr::null_mut();
+        unsafe {
+            bind_ceed::CeedOperatorFieldGetName(self.ptr, &mut name_ptr);
+        }
+        let c_str: &CStr = unsafe { CStr::from_ptr(name_ptr) };
+        let str_slice: &str = c_str.to_str().unwrap().clone();
+        str_slice.to_owned()
+    }
```

Comment:
Should we return a `&str`?

Reply:
I'm not sure why we would? `String` seems to be more common in Rust code than `&str`

Reply:
Standard practice [C-CALLER-CONTROL](https://rust-lang.github.io/api-guidelines/flexibility.html#caller-decides-where-to-copy-and-place-data-c-caller-control) (somewhat different context) says one should not clone unnecessarily inside library interfaces. I think the caller would typically match on the return, use it as a key, or interpolate it into a new string (for end-user output or to create a new related string). They can always call `to_owned()` if they want, but I think it makes more sense for us to return `&str`.